src/odemis/util/polar.py

```diff
@@ -66,81 +56,138 @@ def AngleResolved2Polar(data, output_size, hole=True, dtype=None):
 
+    # populate arrays with values
     for i in xrange(image_x):
-        ypix = (i - mirror_y) + (2 * parabola_f) / pixel_size[1]
+        ypix = (i - mirror_y) + (2 * parabola_f) / pixel_size[1]  # y coordinates of the pixels
+
+        # TODO can we modify xpix and ypix so we only search for angles within half circle
+        # TODO skip those angles outside (e.g. numpy.nan)
+        # Finds the angles of emission by the sample for each px (x,y) in the raw data.
+        # These angles only depend on the mirror geometry.
+        # Each px in the raw data corresponds to a specific theta-phi combination
         theta, phi, omega = _FindAngle(data, xpix, ypix, pixel_size)
 
+        # theta_data: array containing theta values for each px in raw data
         theta_data[i, :] = theta
+        # phi_data: array containing phi values for each px in raw data
         phi_data[i, :] = phi
+
+        # omega_data contains the intensity values from raw data.
+        # It already reflects the shape of the mirror
+        # and is normalized by omega (solid angle:
+        # measure for photon collection efficiency depending on theta and phi)
         omega_data[i, :] = cropped_image[i] / omega
 
-    # Convert into polar coordinates
+    return theta_data, phi_data, omega_data, circle_mask_dilated
+
+
+def AngleResolved2Polar(data, output_size, hole=True, dtype=None):
+    """
+    Converts an angle resolved image to polar (aka azimuthal) projection
+    data (model.DataArray): The image that was projected on the CCD after being
+      reflected on the parabolic mirror. The flat line of the D shape is
+      expected to be horizontal, at the top. It needs PIXEL_SIZE and AR_POLE
+      metadata. Pixel size is the sensor pixel size * binning / magnification.
+    output_size (int): The size of the output DataArray (assumed to be square)
+    hole (boolean): Crop the pole if True
+    dtype (numpy dtype): intermediary dtype for computing the theta/phi data
+    returns (model.DataArray): converted image in polar view
+    """
+
+    # calculate the corresponding theta and phi angles based on the geometrical properties
+    # of the mirror for each px on the raw data
+    # theta_data: array containing theta values for each px in raw data
+    # phi_data: array containing phi values for each px in raw data
+    # omega_data: array containing the measured intensity values for each px in raw data,
+    # which are corrected for photon collection efficiency
+    theta_data, phi_data, omega_data, circle_mask_dilated =\
+        calcAngles(data, hole, dtype)
+
+    # Crop the raw input data based on the mirror mask (circle_mask)
+    # to save memory and improve runtime.
+    # We use a dilated mask for cropping to avoid edge effects during triangulation.
+    # The additional data points (due to dilation) will be set to zero during the interpolation step by omega_data.
+    theta_data_masked = numpy.where(circle_mask_dilated, theta_data, numpy.nan)
```

Comment:
This looks awfully complicated. You take the masked points by NaN, then look where the NaN are in the data, then inverse it, and then take only these points in the data? I think you should be able to just write:
`theta_data_masked = theta_data[circle_mask_dilated]`